### ISSUE TITLE
Benchmark should use a time source with sub-second resolution

### ISSUE THREAD

**Issue by mejedi** [shadowofneptune/threaded-code-benchmark#1]
The benchmark uses `time()` from `time.h` to measure elapsed time. It measures time with 1 second resolution. As certain runs complete in 6 seconds or less, the not so fine grained time source introduces significant inaccuracy.

**Comment by shadowofneptune** [comment]
I agree. I had chosen time() due to its portability, but it's not as good as it could be. Which timekeeping system would you recommend?
